read-json.js

```diff
@@ -329,12 +329,18 @@ function githead (file, data, cb) {
   var dir = path.dirname(file)
   var head = path.resolve(dir, '.git/HEAD')
   fs.readFile(head, 'utf8', function (er, head) {
-    if (er) return cb(null, data)
-    githead_(file, data, dir, head, cb)
+    if (er) {
+      var parent = path.dirname(dir)
+      if (parent === dir) {
+        return cb(null, data)
+      }
+      return githead(dir, data, cb)
+    }
```

Comment:
Shouldn't this be `githead_`?

Reply:
I don't think so - `githead()` reads the `.git/HEAD` file (if found) and passes the content to `githead_()`. So to traverse up the tree and find the `.git/HEAD` file, we need to call `githead()` recursively.
`githead_()` is what then handles the contents of the `.git/HEAD` file by reading the ref it points to etc.

It's easy to get confused though with the naming of the functions and no comments...